src/components/accession2/viabilityTesting/NewViabilityTestModal.tsx

```diff
@@ -166,6 +160,27 @@ export default function NewViabilityTestModal(props: NewViabilityTestModalProps)
     }
   };
 
+  const calculatNewTotal = (id: string, value: string) => {
+    if (record) {
+      switch (id) {
+        case 'seedsCompromised':
+          return (Number(record.seedsEmpty) || 0) + (Number(record.seedsFilled) || 0) + Number(value);
+        case 'seedsFilled':
+          return (Number(record.seedsEmpty) || 0) + (Number(record.seedsCompromised) || 0) + Number(value);
+        default:
+          return (Number(record.seedsFilled) || 0) + (Number(record.seedsCompromised) || 0) + Number(value);
+      }
+    }
+  };
+
+  const onChangeCutValue = (id: string, value: unknown) => {
+    if (record && typeof value === 'string') {
+      const newTotal = calculatNewTotal(id, value);
+      setRecord({ ...record, [id]: value, seedsTested: newTotal || 0 });
```

Comment:
do we need to test for NaN?

```
const caculatedTotal = calculateNewTotal(id, value);
const newTotal = isNaN(calculatedTotal) ? 0 : calculatedtotal;
setRecord({ ...record, [id]: value, seedsTested: newTotal });
setTotalSeedsTested(newTotal);
```

Reply:
if we can follow up on this at some point might be good, noticed that our textfields are not of type number .. we can edit characters and ui accepts it.. BE errors out. maybe when you add validation for this form.